src/main/groovy/com/dicedmelon/gradle/jacoco/android/JacocoAndroidPlugin.groovy

```diff
@@ -64,15 +68,24 @@ class JacocoAndroidPlugin implements Plugin<ProjectInternal> {
     def classesDir = classesDir(variant)
     def testTask = testTask(project.tasks, variant)
     def executionData = executionDataFile(testTask)
+    def kotlin = hasKotlin(project.plugins)
     JacocoReport reportTask = project.tasks.create("jacoco${testTask.name.capitalize()}Report",
         JacocoReport)
     reportTask.dependsOn testTask
     reportTask.group = "Reporting"
     reportTask.description = "Generates Jacoco coverage reports for the ${variant.name} variant."
     reportTask.executionData = project.files(executionData)
     reportTask.sourceDirectories = project.files(sourceDirs)
-    reportTask.classDirectories =
+    def javaTree =
         project.fileTree(dir: classesDir, excludes: project.jacocoAndroidUnitTestReport.excludes)
+    if (kotlin) {
```

Comment:
This if-else statement might cause some problems in mixed Java/Kotlin projects. Don't you think?

Reply:
Would it? It uses both java and kotlin trees. Unless the same classes up in both places?

Reply:
Our project is mixed and I haven't noticed any issues